src/set.rs

```diff
@@ -380,6 +380,31 @@ fn example() -> Result<(), Box<dyn std::error::Error>> {
     pub fn into_fst(self) -> raw::Fst<D> {
         self.0
     }
+
+    /// Maps the underlying data of the fst Set to another data type.
+    ///
+    /// # Example
+    ///
+    /// This example shows that you can map an fst Set based on a `Vec<u8>`
+    /// into an fst Set based on a `Cow<[u8]>`, it can also work with a
+    /// reference counted type (e.g. `Arc`, `Rc`)
+    ///
+    /// ```rust
+    /// use std::borrow::Cow;
+    /// use fst::Set;
+    ///
+    /// let set: Set<Vec<u8>> = Set::from_iter(&["hello", "world"]).unwrap();
+    ///
+    /// let set_on_cow: Set<Cow<[u8]>> = set.map_data(Cow::Owned).unwrap();
+    /// ```
```

Comment:
See comments above on the `Map` example and apply them to this as well. :-) Thanks.